api/src/main/java/io/strimzi/api/kafka/model/KafkaAuthorizationCustom.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.DescriptionFile;
+import io.strimzi.crdgenerator.annotations.Example;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+
+/**
+ * Configures the broker authorization
+ */
+@DescriptionFile
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"type", "implementionClass", "superUsers"})
+@EqualsAndHashCode
+public class KafkaAuthorizationCustom extends KafkaAuthorization {
```

Comment:
I think we will need to add some `config` section here. I think that would need to be passed to the user configuration section.

Reply:
What I have seen with other types in `KafkaBrokerConfigurationBuilder` is passing the key/value pairs, and I believe they get converted `configure(Map<String, Object> config)` method on `Authorizer` class. Since we can not send a complex type as I suspect this content is being serialized, thus I was saying yaml/.properties contents but was afraid of any escaping rules of the content.

Reply:
`Map<String, Object>` is already used on many places through the code ... just look at `.spec.kafka.config` for example. You do not need to deal with any properties or anything.

Reply:
ok, Thanks.  I see there is already `addtionalProperties` with type `Map<String, Object>` in the base class, I will introduce it as a `configuration` section and pass on the properties. so, basically would look like

```
authorization:
  type: custom
  authorizerClass: io.mycompany.CustomAuthorizer
  configuration:
    prop1=value1
    prop2=value2
  superUsers:
    - CN=client_1
    - user_2
    - CN=client_3
```

Reply:
The additional properties have to stay as a separate field. That is used for any fields which would be unknown. Also, the YAML will look like this at the end:

```yaml
authorization:
  type: custom
  authorizerClass: io.mycompany.CustomAuthorizer
  configuration:
    prop1: value1
    prop2: value2
  superUsers:
    - CN=client_1
    - user_2
    - CN=client_3
```